src/Hazelcast.Net/Clustering/ClusterMembers.cs

```diff
@@ -80,6 +80,12 @@ public ClusterMembers(ClusterState clusterState, TerminateConnections terminateC
                     lock (_mutex) return _members.ContainsMember(x);
                 }, _clusterState.LoggerFactory);
             }
+
+            _clusterState.Failover.ClusterChanged += cluster =>
+            {
+                // Clear old member table since failover happened.
+                _members = new MemberTable();
```

Comment:
Member removed events may be missing during cluster switch, let's have a look at this.

Reply:
Great catch! I've covered that part at https://github.com/hazelcast/hazelcast-csharp-client/pull/891/commits/285b613d5b6de629b9ac9d086637dc8d6a839e98